src/libcore/slice/mod.rs

```diff
@@ -80,7 +80,7 @@ macro_rules! slice_offset {
     ($ptr:expr, $by:expr) => {{
         let ptr = $ptr;
         if size_from_ptr(ptr) == 0 {
-            (ptr as *mut i8).wrapping_offset($by) as _
+            (ptr as *mut i8).wrapping_offset($by.wrapping_mul(align_from_ptr(ptr) as isize)) as _
```

Comment:
Is there enough address space to do this, given the capacities claimed by `RawVec`?  I can apparently make a `usize::MAX`-length `Vec` of a align-8 ZST: https://play.rust-lang.org/?gist=c0b3d266bfbd34448d97e00f1408c946&version=stable&mode=release&edition=2015

Should the following change to `!0 / mem::align_of::<T>()`?

https://github.com/rust-lang/rust/blob/2c1a715cbda1d6eba39625aca08f1f2ac7c0dcc8/src/liballoc/raw_vec.rs#L215-L221

Reply:
Ah, good point.

Is it enough to change that in `raw_vec` or are there other ways to create huge slices?

Reply:
Yes it does. In fact I am working on that next step already. I thought it would be easier for reviewers etc. to split this into two PRs, but if you prefer I can certainly finish that up and add it here (or make a new PR because there are going to be drastically more changes).